conftest.py

```diff
@@ -7,7 +7,7 @@
 
 from topo_processor.stac.iter_errors_validator import IterErrorsValidator
 
-# pystac.validation.set_validator(IterErrorsValidator())
+pystac.validation.set_validator(IterErrorsValidator())
```

Comment:
Using a `fixture` with the `autouse=True` (`@pytest.fixture(autouse=True)`) will set this validator before each test.
```suggestion
@pytest.fixture(autouse=True)
def set_iter_errors_validator() -> None:
 pystac.validation.set_validator(IterErrorsValidator())
```

Check out [Autouse fixture](https://docs.pytest.org/en/6.2.x/fixture.html#autouse-fixtures-fixtures-you-don-t-have-to-request)

When you need to use another validator you could hardcode it at the beginning of the test.
That would avoid you to set back the `IterErrorsValidator` at the end of the `test_check_multiple_stac_extensions_default_pystac_validator()` test too.